app/lib/metrics_service.rb

```diff
@@ -89,13 +101,14 @@ def self.log(name, props={}, user)
     headers = get_default_headers(user)
 
     access_token = user.token_for_api_call
-    user_id = user.id
+    user_id = user.get_metrics_uuid
 
-    if access_token.nil?
+    if access_token.nil? || !user.registered_for_firecloud
       # User is unauthenticated / unregistered / anonynmous
       props['distinct_id'] = user_id
       headers.delete('Authorization')
       props['authenticated'] = false
+      props['registeredForTerra'] = false
```

Comment:
same question about whether we want another property here about whether the user is signed into SCP